src/plugins/data/public/ui/query_editor/language_selector.tsx

```diff
@@ -3,16 +3,25 @@
  * SPDX-License-Identifier: Apache-2.0
  */
 
-import { EuiComboBox, EuiComboBoxOptionOption, PopoverAnchorPosition } from '@elastic/eui';
+import {
+  EuiComboBoxOptionOption,
+  PopoverAnchorPosition,
+  EuiContextMenuPanel,
+  EuiPopover,
+  EuiButtonEmpty,
+  EuiContextMenuItem,
+} from '@elastic/eui';
 import { i18n } from '@osd/i18n';
-import React from 'react';
+import React, { useState } from 'react';
 import { getUiService } from '../../services';
 
-interface Props {
+export interface QueryLanguageSelectorProps {
   language: string;
   onSelectLanguage: (newLanguage: string) => void;
   anchorPosition?: PopoverAnchorPosition;
   appName?: string;
+  languageSelectorContainerRef?: React.RefCallback<HTMLDivElement>;
+  noArrow?: boolean;
```

Comment:
nit: i think we should just get rid of the arrow all together right?

Reply:
removed